up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -41,331 +41,300 @@ service uDiscovery {
-    MESSAGE = 6;      // uE Message
-    RESOURCE = 7;     // uE Resource
-    USER = 8;         // User Information
+  rpc GetuETopics(GetuETopicsRequests) returns (GetuETopicsResponse){
+    option (method_id) = 2;
   }
-}
 
+  /* 
+  Consumers shall call GetUEList API to fetch the list of uEs under a given device.
+  Idea is to provide an API(at each levle in hierarchy) to get a list of local entities.
+  For example: uDDS might call on uLDS for each device to build up its local tree in case of failures.
+  */
 
-// Message passed to the UpdateNode() RPC call
-message UpdateNodeRequest {
-  // Node to be updated in the database
-  Node node = 1;
-
-  // Time-to-live: How long (ms) the information should be live for in the database.
-  // -1 means lives forever
-  optional int32 ttl = 3;
-}
+  /* 
+  Example: 
+  GetUETopics(“up:/device1”)
+  Response: [“up:/device1/uE1”, “up:/device1/uE2”], Status: code.OK 
 
+  GetUETopics(“up:/device2”)
+  Response: [], Status: code.INVALID_ARGUMENT (passed device should match local device)
+  */
 
-// Delete one or more nodes request
-message DeleteNodesRequest {
-  repeated string uris = 1;   // uProtocol formatted URI
-}
+  rpc GetuEList(GetuEListRequest) returns (GetuEListResponse){
+    option (method_id) = 4;
+  }
 
 
-// FindNodesRequest passed to FindNodes()
-message FindNodesRequest {
-  // uProtocol formatted URI for the node to search, ex. '//vcu.VIN/core.app.hartley'
-  // shall return the 'core.app.hartley' node
-  string uri = 1;
+ /*
+  Consumers shall call GetDeviceList API to fetch the list of uEs under a given domain.
+  This is an extension of GetUEList at a higher level.
+  For example: uCDS might call on uDDS to build up its local tree in case of failures
+ */
 
-  // How deep in the node tree should the results return. A value of -1 or
-  // the field is not present means all child nodes are returned, value of 0
-  // returns only the parent node, any other value is the depth of child nodes
-  optional int32 depth = 2;
-}
+ /* Example: 
+  GetDeviceList(“up:/domain”)
+  Response: [“up:/domain/device1”, “up:/domain/device2”], Status: code.OK 
 
+  GetUETopics(“up:/domain2”)
+  Response: [], Status: code.INVALID_ARGUMENT (invalid domain)
 
-// FindNodesResponse that is returned from the FindNodes() API
-message FindNodesResponse {
-  // List of node information
-  repeated Node nodes = 1;
+  */
+  rpc GetDeviceList(GetDeviceListRequest) returns (GetDeviceListResponse){
+    option (method_id) = 5;
+  }
 
-  // Return code for the rpc call
-  uprotocol.v1.UStatus status = 2;
+  /******************************* Data replication / synchronization ******/
 
-  // Time-to-live: How long (ms) the information should be live for in the database.
-  // -1 means lives forever
-  optional int32 ttl = 3;
-}
+  /*
+    UpdateData is NOT a customer facing API. This shall be used discovery 
+    instances to push an update between to core components (streamer)/ other discovery instances.
```

Comment:
Then do not put this here. Also Streamer should not be updating information in the discovery database.